### DISCUSSION TITLE
ScriptReference/Vector3.Distance

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#41] [Notes]
# ScriptReference/Vector3.Distance

https://docs.unity3d.com/ScriptReference/Vector3.Distance.html

**Comment by vertxxyz** [comment]
Use [`Vector3.sqrMagnitude`](https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html) when comparing distances[^1] to avoid the square root operation present in the distance calculation.
Performance optimisations like these may only be necessary in hot code paths.
[^1]: When comparing distances, the same one is longer even when both are squared. When you already have another distance squaring (`a * a`) it will still be faster than a square root.
